### PR TITLE
Extend discovery based on Medina issues

### PR DESCRIPTION
**Pull request by anatheka**
This PR updates/adds the following discovered resources properties:
- Update function runtime/version 
- Fix OsLoggingEnabled
- Add ActivityLoggingEnabled

### CODE REVIEW

File: service/discovery/azure/compute.go [hunk 1]
```diff
@@ -272,6 +342,18 @@ func (d *azureComputeDiscovery) handleVirtualMachines(vm *armcompute.VirtualMach
...
 	}
 
+	// Check extensions
+	for _, extension := range vm.Resources {
+		// Azure Monitor Agent (AMA) collects monitoring data from the guest operating system of Azure and hybrid virtual machines and delivers it to Azure Monitor for use (https://learn.microsoft.com/en-us/azure/azure-monitor/agents/agents-overview). The extension names are
+		// * OMSAgentForLinux for Linux VMs and (legacy agent)
+		// * MicrosoftMonitoringAgent for Windows VMs (legacy agent)
+		// * AzureMonitoringWindowsAgent (new agent)
+		// * AzureMonitoringLinuxAgent (new agent)
+		if strings.Contains(*extension.ID, "OmsAgentForLinux") || strings.Contains(*extension.ID, "MicrosoftMonitoringAgent") || strings.Contains(*extension.ID, "AzureMonitoringWindowsAgent") || strings.Contains(*extension.ID, "AzureMonitoringLinuxAgent") {
```
Comment (lebogg) [inline service/discovery/azure/compute.go:352]: I guess there are no constants in the azure sdk defining these strings?

File: service/discovery/azure/compute.go [hunk 2]
```diff
@@ -303,16 +385,23 @@ func (d *azureComputeDiscovery) handleVirtualMachines(vm *armcompute.VirtualMach
...
 					SecurityFeature: &voc.SecurityFeature{},
 				},
 				MonitoringLogDataEnabled: monitoringLogDataEnabled,
 				SecurityAlertsEnabled:    monitoringLogDataEnabled,
 			},
 		},
+		ActivityLogging: &voc.ActivityLogging{
+			Logging: &voc.Logging{
+				Enabled:         true,               // is always enabled
+				RetentionPeriod: 90,                 // always 90 days
```
Comment (lebogg) [inline service/discovery/azure/compute.go:401]: Maybe consider using a constant for retentionperiod

File: service/discovery/azure/compute.go [hunk 3]
```diff
@@ -123,41 +123,71 @@ func (d *azureComputeDiscovery) List() (list []voc.IsCloudResource, err error) {
...
+			var r voc.IsCompute
+
+			// Check kind of site (see https://github.com/Azure/app-service-linux-docs/blob/master/Things_You_Should_Know/kind_property.md)
+			switch *site.Kind {
+			case "app": // Windows Web App
+			// TODO(all): TBD
+			case "app,linux": // Linux Web app
+			// TODO(all): TBD
+			case "app,linux,container": // Linux Container Web app
+			// TODO(all): TBD
```
Comment (lebogg) [inline service/discovery/azure/compute.go:167]: Add test cases for these, maybe to check that `r` is nil/empty

File: service/discovery/azure/compute.go [hunk 4]
```diff
@@ -123,41 +123,71 @@ func (d *azureComputeDiscovery) List() (list []voc.IsCloudResource, err error) {
 	}
 	list = append(list, virtualMachines...)
 
-	// Discover functions
-	function, err := d.discoverFunctions()
```
[the remark this answers is not in the document]
Comment (anatheka) [reply service/discovery/azure/compute.go:127]: I'm not sure. At the end we discover functions and web apps and sites is only a step in between.

File: service/discovery/azure/compute.go (hunk 1 above)
Comment (anatheka) [reply service/discovery/azure/compute.go:352]: No.

File: service/discovery/azure/compute.go (hunk 3 above)
Comment (anatheka) [reply service/discovery/azure/compute.go:167]: That is a log of overhead because I have to add a function for each case in the Do method. I think that is out of proportion and ok if we only check 2 cases.

File: service/discovery/azure/compute.go [hunk 5]
```diff
@@ -123,41 +123,71 @@ func (d *azureComputeDiscovery) List() (list []voc.IsCloudResource, err error) {
...
+			// TODO(all): TBD
+			case "functionapp": // Function Code App
+				r = d.handleFunction(site)
+			case "functionapp,linux": // Linux Consumption Function app
+				r = d.handleFunction(site)
+			case "functionapp,linux,container,kubernetes": // Function Container App on ARC
+			// TODO(all): TBD
+			case "functionapp,linux,kubernetes": // Function Code App on ARC
+				// TODO(all): TBD
+			}
```
[the remark this answers is not in the document]
Comment (anatheka) [reply service/discovery/azure/compute.go:184]: I added a default case with a debug message.  If we are in the default case it is not unnecessarily an error, so I think the log message is enough.

File: service/discovery/azure/compute.go [hunk 6]
```diff
@@ -123,41 +123,71 @@ func (d *azureComputeDiscovery) List() (list []voc.IsCloudResource, err error) {
...
-	function, err := d.discoverFunctions()
+	// Discover functions and web apps
+	function, err := d.discoverSites()
 	if err != nil {
 		return nil, fmt.Errorf("could not discover functions: %w", err)
 	}
-	list = append(list, function...)
+	if function != nil {
+		list = append(list, function...)
+	}
```
[the remark this answers is not in the document]
Comment (anatheka) [reply service/discovery/azure/compute.go:133]: This is a relic. :)

**Review by lebogg — APPROVED** [review]
